mesonbuild/dependencies/pkgconfig.py

```diff
@@ -30,14 +39,24 @@ class PkgConfigInterface:
 
     class_impl: PerMachine[T.Union[Literal[False], T.Optional[PkgConfigInterface]]] = PerMachine(False, False)
     class_cli_impl: PerMachine[T.Union[Literal[False], T.Optional[PkgConfigCLI]]] = PerMachine(False, False)
+    pkg_bin_per_machine: PerMachine[T.Optional[ExternalProgram]] = PerMachine(None, None)
+
+    @staticmethod
+    def set_program_override(pkg_bin: ExternalProgram, for_machine: MachineChoice) -> None:
+        if PkgConfigInterface.class_impl[for_machine]:
+            raise MesonException(
+                f"Tried to override pkg-config for machine {for_machine} but it was already initialized.\n"
+                "pkg-config must be overridden before it's used."
+            )
```

Comment:
Existing code style tends to favor:
```python
raise MesonException(long_string_part_1,
                     long_string_part_2)
```

We are pretty consistent about using single quotes, not double quotes.